integration-tests/olminstall/steps/tekton_incluster.py

```
@@ -274,3 +277,203 @@ def list_taskruns_in_cluster(
+def fetch_snapshot_metadata(
+    snapshot_name: str,
+    namespace: str | None = None,
+) -> tuple[dict[str, str], dict[str, str]]:
+    """Return Snapshot ``metadata.labels`` and ``metadata.annotations`` as string dicts."""
+    snap = (snapshot_name or "").strip()
+    ns = (namespace or namespace_from_env()).strip()
+    if not snap or not ns:
+        return {}, {}
+
+    doc: dict[str, Any] | None = None
+    proc = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)